### ISSUE TITLE
Cannot supply THREAD_OPTIONS to lzmq.threads

### ISSUE THREAD

**Issue by johnfoconnor** [zeromq/lzmq#28]
there should be an api form which allows the user to supply their own THREAD_OPTIONS

something like

```lua
function zthreads.fork(ctx, code, ...)
  local fork_starter = {code = code} 
  return zthreads.fork_opts(ctx, fork_starter, ...)
end

function zthreads.fork_opts(ctx, thread_opts, ...)
  local pipe, endpoint = make_pipe(ctx)
  if not pipe then return nil, endpoint end

  ctx = ctx:lightuserdata()
  local ok, err = Threads.new(thread_opts, ZMQ_NAME, ctx, endpoint, ...)
  if not ok then
    pipe:close()
    return nil, err
  end
  return ok, pipe
end
```

use case (supplying package paths to child thread)

```lua
loca lua_init = [[
  package.cpath="]] .. package.cpath .. [["
  package.path="]] .. package.path
local opts = {lua_init = lua_init, code = "@my_file.lua"}
zthreads. fork_opts(ctx, opts)
```

**Comment by johnfoconnor** [comment]
im willing to implement this. Do you have input on what the api should look like?

i'm suggesting

zthreads.run_opts(ctx, thread_options, ...)
zthreads.fork_opts(ctx, thread_options, ...)
zthreads.actor_opts(ctx, thread_options, ...)

where thread_options is
https://github.com/moteus/lua-llthreads2/blob/master/src/lua/llthreads2/ex.lua#L196

**Comment by moteus** [comment]
Closed by #29 and #30
